### ISSUE TITLE
Question about learning process and paper

### ISSUE THREAD

**Issue by h19920918** [google-research/text-to-text-transfer-transformer#164]
@craffel Thank you for your answer.

Actually, it means suppose that

The original text is "Thank you for your answer"

In the step 1, ("your" is selected to mask)
Input: Thank you for (X) answer (EOS)
Target: (BOS) (X) your (Y) (EOS)

And, in the another step N, ("for" is selected to mask)
Input: Thank you (X) your answer (EOS)
Target: (BOS) (X) for (Y) (EOS)

In this case, (X) learned the representation from the "for" and "your" given sentence.
But in the generation process, given (X) what will be next predicted token?
Because (X) learned to predict "for" and "your" in the steps.

The actual meaning of this question is to further pre-train a new context by the unsupervised objective.
When I further pre-trained on the SQuAD context by the unsupervised objective, it showed significant dropping performance.

I suspected occurring this fact from that the pre-trained weights was already pre-trained from the supervised datasets by own label after pre-trained by the unsupervised objective.
So, it has some inference ability by the pre-training with the multi supervised datasets.

For the summary of the second,

The pre-trained weights provided by this repo.,
(Pre-train) C4 with unsupervised objective --> (Pre-train) Multiple tasks with their supervised objective

I want to make the above to this,
(Pre-train) C4 with unsupervised objective --> (Pre-train) Multiple tasks with supervised objective --> (Further pre-train) a new task with unsupervised objective --> (Fine-tune) a new task with supervised objective

When I do this, can we expect performance improvement?
In the SQuAD, it was not.

_Originally posted by @h19920918 in https://github.com/google-research/text-to-text-transfer-transformer/issues/163#issuecomment-611322374_

**Comment by craffel** [comment]
Answered in #163
